zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/ClusterVariableHandler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.handlers;
+
+import io.camunda.exporter.exceptions.PersistenceException;
+import io.camunda.exporter.store.BatchRequest;
+import io.camunda.webapps.schema.entities.clustervariable.ClusterVariableEntity;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.ClusterVariableIntent;
+import io.camunda.zeebe.protocol.record.value.ClusterVariableRecordValue;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+public class ClusterVariableHandler
+    implements ExportHandler<ClusterVariableEntity, ClusterVariableRecordValue> {
+
+  private static final Set<ClusterVariableIntent> SUPPORTED_INTENTS =
+      EnumSet.of(ClusterVariableIntent.CREATED, ClusterVariableIntent.DELETED);
+  private final String indexName;
+  private final int variableSizeThreshold;
+
+  public ClusterVariableHandler(final String indexName, final int variableSizeThreshold) {
+    this.indexName = indexName;
+    this.variableSizeThreshold = variableSizeThreshold;
+  }
+
+  @Override
+  public ValueType getHandledValueType() {
+    return ValueType.CLUSTER_VARIABLE;
+  }
+
+  @Override
+  public Class<ClusterVariableEntity> getEntityType() {
+    return ClusterVariableEntity.class;
+  }
+
+  @Override
+  public boolean handlesRecord(final Record<ClusterVariableRecordValue> record) {
+    return SUPPORTED_INTENTS.contains(record.getIntent());
```

Comment:
```suggestion
    return getHandledValueType().equals(record.getValueType())
        && SUPPORTED_INTENTS.contains(record.getIntent());
```

🔧 Let's make it "safer" - to be fair, not all handlers check for the value type; however, I still recommend adding this check here.